api/mesh/v1alpha1/gateway_route.proto

```diff
@@ -245,10 +245,18 @@ message MeshGatewayRoute {
         ];
       };
 
+      message Rewrite {
+        oneof path {
+          string replace_full = 1;
+          string replace_prefix_match = 2;
```

Comment:
how prefix rewrite works if you're matching by header (not by prefix path)?

Reply:
It only works if you match by prefix AFAIU (actually something to add in validation)